src/web_socket.toit

```diff
@@ -144,9 +145,27 @@ class WebSocket:
         masking_bytes = null
     result := FragmentReader_ this len pending_[0] --masking_bytes=masking_bytes
     if not result.is_ok_:
-      close
+      close --status_code=STATUS_WEBSOCKET_PROTOCOL_ERROR
       throw "PROTOCOL_ERROR"
+
     pending_ = pending_[header_size_needed..]
+
+    if result.is_close:
+      if result.size >= 2:
+        // Two-byte close reason code.
+        payload := #[]
+        while packet := result.read:
+          payload += packet
```

Comment:
Maybe guard against excessive payload?

`if payload.size > 2: throw "PROTOCOL ERROR"` or something?

Reply:
No, that's allowed, but we ignore it.